### ISSUE TITLE
How to generate good image in 1024*1024

### ISSUE THREAD

**Issue by cjb752815359** [junyanz/pytorch-CycleGAN-and-pix2pix#838]
At first, thanks for your work!
Recently, I am using this project to do something like image-to-image translation.
I hope to generate images with the style I want.But the generated images doesn't look good .
I find that when the size of image is small, like 256*256, the result is good.
Maybe I should train the model with the whole image without resize and crop? But my GPU can't afford.

**Comment by junyanz** [comment]
If you have pairs, please refer to our recent work such as [pix2pixHD](https://github.com/NVIDIA/pix2pixHD) and [SPADE](https://github.com/NVlabs/SPADE). If you are using CycleGANs, I recommend that you train the models on cropped patches. Please refer to [training/test tips](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/docs/tips.md#trainingtesting-with-high-res-images) for more details.

**Comment by MuhammadAsadJaved** [comment]
@cjb752815359  

https://github.com/affinelayer/pix2pix-tensorflow/issues/56

**Comment by cjb752815359** [comment]
> If you have pairs, please refer to our recent work such as [pix2pixHD](https://github.com/NVIDIA/pix2pixHD) and [SPADE](https://github.com/NVlabs/SPADE). If you are using CycleGANs, I recommend that you train the models on cropped patches. Please refer to [training/test tips](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/docs/tips.md#trainingtesting-with-high-res-images) for more details.

Thanks for your reply. I am using CycleGAN now. I use the default parameters except load_size=1024, crop_size=400(the maximal I can set) when training, load_size=crop_size=1024 when testing. But the result is not good.
I found that you apply the network in large image, can you tell me the parameters?

**Comment by junyanz** [comment]
We used a similar configuration for GTA5 to cityscapes. See @taesungp's reply at this [post](#586).

**Comment by asheroin** [comment]
> If you have pairs, please refer to our recent work such as [pix2pixHD](https://github.com/NVIDIA/pix2pixHD) and [SPADE](https://github.com/NVlabs/SPADE). If you are using CycleGANs, I recommend that you train the models on cropped patches. Please refer to [training/test tips](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/docs/tips.md#trainingtesting-with-high-res-images) for more details.


Is there a suggested threshold for using HD or just the normal version? In Pix2PixHD paper, the output for global network is 1024 * 512,  just a little smaller than 1024*1024 or,  in other words, has a same width compare with 1024*1024. However, in general application, a 256*256 size of patches would be a more common choice for common pix2pix model which serving as the global network in HD version.  I not sure if a  common pix2pix model could directly handle the 1024 * 1024 or 512 * 512 input size.

Further more, when using the HD version, how to design the output size for global network? Half of the input size of fix it to 1024 or 512 no matter how large the input is? Although this question depends heavily on different tasks but I still hope for a general answer.

**Comment by junyanz** [comment]
If you want to generate 1024+ images, you may want to use HD. I guess pix2pix can handle 256. For 512, you can try both. I think pix2pixHD can support images with different sizes. If you have questions regarding this, please post your question on the pix2pixHD repo.

**Comment by Adrian-1234** [comment]
Hi,

I found this project a couple of days ago.....

First thing I'd like to say is thank you very much for sharing this exciting project, and for continuing to support it.

I have been carrying out some experiments on my CPU only system with various sizes of input, which can be summarised in the following script:

for i in style_monet style_cezanne style_ukiyoe style_vangogh
do
     echo $i
     python3 test.py --dataroot datasets/"$i"/testA-1880 --name "$i"_pretrained --model test  --gpu_ids -1 --no_dropout --preprocess none --load_size 1880 --crop_size 1880 &
     python3 test.py --dataroot datasets/"$i"/testA-1920x1080p --name "$i"_pretrained --model test  --gpu_ids -1 --no_dropout --preprocess none --load_size 1920 --crop_size 1920 &
     python3 test.py --dataroot datasets/"$i"/testA-3840 --name "$i"_pretrained --model test  --gpu_ids -1 --no_dropout --preprocess none --load_size 3840 --crop_size 3840 &
     python3 test.py --dataroot datasets/"$i"/testA-orig-small --name "$i"_pretrained --model test  --gpu_ids -1 --no_dropout &
     wait
done

Where 1880, 1920 and 3840 are the horizontal resolutions  of the input image set.

Not only am I able to process (test/inference) very large images with good results, but your code is parallel (on the CPUs) and runs quickly on my 120 core system. In the above case,  I run the streams in parallel to increase the systems utilisation as much as possible.

Keep up the good work !

**Comment by junyanz** [comment]
Thanks a lot. Good luck with your project.

**Comment by AugustLee93** [comment]
> If you want to generate 1024+ images, you may want to use HD. I guess pix2pix can handle 256. For 512, you can try both. I think pix2pixHD can support images with different sizes. If you have questions regarding this, please post your question on the pix2pixHD repo.

Hi @junyanz ,  recently I try pix2pix for 512, it seems like some generated image missing some detail or it didn't match the groudtruth well, then I try pix2pixHD, it does a good job, but the huge problem is that the model has artifacts randomly generated on image，I mean It training stage, the generate image has nothing artifacts at all. But in testing stage, same testing dataset folder, if I generate multiple times, the result is difference( here for example I mean ,1st gen_img1.jpg has no artifacts, 2nd time generated result (gen_img1.jpg has an artifacts on it as I showed in attached)
![artifacts](https://user-images.githubusercontent.com/70758744/141714464-ae7da7fb-909c-4fde-9d27-278eb817b2a0.png)

So my question will be, do you have any suggestions or advices for me ? like for pix2pix, should I set --num_D= pixel when training? Would that be help to improve some detail generated?  Or gan_loss model default  is lsgan, using vanilla would help?  For artifacts problem, I try 
 the solution in #issue 190, It seems like it didn't change much. Looking forward your professional advices, any helps I will very appreciate it! Thanks

**Comment by junyanz** [comment]
It's hard for pix2pix to produce high-res images. For high-res image synthesis, feel free to try pix2pixHD, [SPADE](https://github.com/NVlabs/SPADE), [SEAN](https://arxiv.org/abs/1911.12861), or other recent models. 
In pix2pixHD, maybe you can increase the weight of L1 loss and VGG loss to make the model more conservative. 

I am not familiar with stochasticity during test time. Maybe you want to check if you can produce the same result with the same random seed or not.

**Comment by AugustLee93** [comment]
> 

Thank you for your useful reply, I will try it later. Thanks again!

**Comment by AugustLee93** [comment]
> It's hard for pix2pix to produce high-res images. For high-res image synthesis, feel free to try pix2pixHD, [SPADE](https://github.com/NVlabs/SPADE), [SEAN](https://arxiv.org/abs/1911.12861), or other recent models. In pix2pixHD, maybe you can increase the weight of L1 loss and VGG loss to make the model more conservative.
> 
> I am not familiar with stochasticity during test time. Maybe you want to check if you can produce the same result with the same random seed or not.

Thank you for your useful reply, I will try it later. Thanks again!
